### PR TITLE
Remove `icu_datagen` legacy API

### PR DESCRIPTION
**Pull request by robertbastian**
Based on #4960

### CODE REVIEW

File: provider/datagen/src/bin/icu4x-datagen.rs [hunk 1]
```diff
@@ -132,34 +132,17 @@ struct Cli {
...
-    key_file: Option<PathBuf>,
-
     #[arg(long, value_name = "BINARY")]
     #[arg(help = "Analyzes the binary and only includes keys that are used by the binary.")]
     keys_for_bin: Option<PathBuf>,
 
-    #[arg(long, hide = true)]
-    #[arg(help = "Deprecated: alias for --keys all")]
-    all_keys: bool,
-
```
Comment (sffc) [inline provider/datagen/src/bin/icu4x-datagen.rs:162]: Nit: We've been piloting the new names for a while, but I still use `--all-keys --all-locales` because I find them easier to remember than `--keys all --locales full`. Please pull these two specific removals to their own PR

**Review by sffc — COMMENTED** [review]
Good! There's just one pair of deprecated APIs that I think we may want to un-deprecate

**Review by sffc — DISMISSED** [review]
I'll approve it to unblock work and we can add the other names back if desired

**Review by Manishearth — DISMISSED** [review]

**Comment by robertbastian** [comment]
I guess it's just `--without-fallback`

**Review by sffc — DISMISSED** [review]

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Comment by robertbastian** [comment]
#2856
